### PR TITLE
changes a few lines of code in the block code of the if satement with…

### PR DESCRIPTION
**Pull request by Jbsmall17**
… conditions:req.url.startsWith('/v1/students/') && req.method === 'PATCH', in the handleRequest function

### CODE REVIEW

**Comment by Jbsmall17** [comment]
The response of the http PATCH METHOD of the api server built in api.js was meant to return the updated version on the students rather it return the un-updated version of the students array. I have resolved that by applying  splice method on the students array(line 73 of api.js file).

File: api.js [hunk 1]
```diff
@@ -69,9 +69,10 @@ const handleRequest = (req, res) => { // handles all requests, routing and metho
         }
 
 
-        const student = { ...students[studentIndex], ...req.body } // updates the student information
-
-        return response({ data: student, code: 200 }) // return response
+        const student = { ...students[studentIndex],id: Math.floor(Math.random() * 10).toString()} //, updates the student information
```
Comment (daniel-alts) [inline api.js:72]: This only changes the id of the student. What if body of the request contains other details?.

File: api.js [hunk 2]
```diff
@@ -69,9 +69,10 @@ const handleRequest = (req, res) => { // handles all requests, routing and metho
...
+        students.splice(studentIndex,1,student) //removes the oldstudent info and replaced it with updated info
```
Comment (daniel-alts) [inline api.js:73]: Adding this here fixes the issue. Awesome work on noticing this bug

**Review by daniel-alts — CHANGES_REQUESTED** [review]
Looks good to me. 

Just update the code to ensure that updates are captured for other properties of the students. No need to update the id

**Comment by Jbsmall17** [comment]
thanks for your review and feedback

**Pull request merged**